specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-11-01-preview/TuningOptions.json

```diff
@@ -0,0 +1,456 @@
+        },
+        "unit": {
+          "type": "string",
+          "description": "Dimension unit"
+        },
+        "queryId": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Optional property that can be used to store the QueryId if the metric is for a specific query."
+        },
+        "absoluteValue": {
+          "type": "number",
+          "format": "double",
+          "description": "Absolute value"
+        }
+      },
+      "description": "Stores property that features impact on some metric if this recommended action is applied."
+    },
+    "IndexRecommendationDetails": {
+      "type": "object",
+      "properties": {
+        "databaseName": {
+          "type": "string",
+          "description": "Database name."
+        },
+        "schema": {
+          "type": "string",
+          "description": "Schema name."
+        },
+        "table": {
+          "type": "string",
+          "description": "Table name."
+        },
+        "indexType": {
+          "type": "string",
+          "description": "Index type."
+        },
+        "indexName": {
+          "type": "string",
+          "description": "Index name."
+        },
+        "indexColumns": {
+          "type": "string",
+          "description": "Index columns. These will be semi-column separated."
+        },
+        "includedColumns": {
+          "type": "string",
+          "description": "Index included columns. These will be semi-column separated."
+        }
+      },
+      "description": "Recommendation details for the recommended action."
+    },
+    "IndexRecommendationResourceProperties": {
+      "type": "object",
+      "properties": {
+        "initialRecommendedTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Creation time of this recommendation in UTC date-time string format."
+        },
+        "lastRecommendedTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The last refresh of this recommendation in UTC date-time string format."
+        },
+        "timesRecommended": {
+          "format": "int32",
+          "type": "integer",
+          "description": "The number of times this recommendation has encountered."
+        },
+        "improvedQueryIds": {
+          "type": "array",
+          "items": {
+            "type": "integer",
+            "format": "int64"
+          },
+          "description": "The ImprovedQueryIds. The list will only be populated for CREATE INDEX recommendations."
+        },
+        "recommendationReason": {
+          "type": "string",
+          "description": "Reason for this recommendation."
+        },
+        "recommendationType": {
+          "type": "string",
+          "description": "Type for this recommendation."
+        },
+        "implementationDetails": {
+          "type": "object",
+          "properties": {
+            "method": {
```

Comment:
Can this be modeled as string enum with a fixed set of values?

Reply:
We do not have a fixed set of values for this property as of yet.